pkg/agent/nodeportlocal/nplagent_test.go

```diff
@@ -127,6 +127,11 @@ func TestMain(t *testing.T) {
 
 	go c.Run(stopCh)
 	informerFactory.Start(stopCh)
+
+	// Must wait for cache sync, otherwise resource creation events will be missing if the resources are created
+	// in-between list and watch call of an informer. This is because fake clientset doesn't support watching with
+	// resourceVersion. A watcher of fake clientset only gets events that happen after the watcher is created.
+	informerFactory.WaitForCacheSync(stopCh)
```

Comment:
I just realized that `TestMain` is not actually a test case but a setup function for other test cases. I think this is not the right way to run setup code. Golang has a [TestMain](https://golang.org/pkg/testing/#hdr-Mai) can do setup and teardown but not like this (argument type is wrong and it didn't call `m.Run()` to run wrapped tests).
The current code results in that the current `TestMain` must run before other tests, and other tests relies on the side effect of it, which means tests cannot be run individually.
Besides, a test should clean up its side effect before finished so that other tests wouldn't be affected by accident, otherwise there will be some weird flakes when more tests are added.

I think you can have two separate setup and teardown functions, build a new NPLController in each test to avoid side effect, unset the env vars in teardown function, e.g.

Reply:
yes, I noticed this but thought it would require more changes that are out of scope of the issue. But sure, I have  added the required changes in the following commit. You can see if you want to squash it eventually.

Reply:
I opened a PR to address these issues: https://github.com/vmware-tanzu/antrea/pull/1826